runtime/src/test/java/com/flipkart/flux/integration/E2ETest.java

```diff
@@ -14,76 +14,105 @@
-import com.flipkart.flux.impl.boot.TaskModule;
-import com.flipkart.flux.initializer.OrderedComponentBooter;
+import com.flipkart.flux.guice.module.*;
+import com.flipkart.flux.initializer.ExecutionOrderedComponentBooter;
+import com.flipkart.flux.initializer.OrchestrationOrderedComponentBooter;
 import com.flipkart.flux.module.DeploymentUnitTestModule;
 import com.flipkart.flux.module.RuntimeTestModule;
 import com.flipkart.flux.registry.TaskExecutableImpl;
+import com.flipkart.flux.registry.TaskExecutableRegistryImpl;
 import com.flipkart.flux.rule.DbClearRule;
 import com.flipkart.flux.runner.GuiceJunit4Runner;
 import com.flipkart.flux.runner.Modules;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import javax.inject.Inject;
 import java.util.Set;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 @RunWith(GuiceJunit4Runner.class)
-@Modules({DeploymentUnitTestModule.class,ShardModule.class,RuntimeTestModule.class,ContainerModule.class,AkkaModule.class,TaskModule.class,FluxClientInterceptorModule.class})
+@Modules(orchestrationModules = {ShardModule.class, RuntimeTestModule.class, ContainerModule.class,
+        OrchestrationTaskModule.class, FluxClientInterceptorModule.class},
+        executionModules = { DeploymentUnitTestModule.class, AkkaModule.class, ExecutionTaskModule.class, ExecutionContainerModule.class, FluxClientInterceptorModule.class})
 public class E2ETest {
 
-    @Inject
+    @InjectFromRole(value = FluxRole.ORCHESTRATION)
     StateMachinesDAO stateMachinesDAO;
 
-    @Inject
+    @InjectFromRole(value = FluxRole.ORCHESTRATION)
     ParallelScatterGatherQueryHelper parallelScatterGatherQueryHelper;
 
-    @Inject
+    @InjectFromRole(value = FluxRole.ORCHESTRATION)
     EventsDAO eventsDAO;
 
     @Rule
-    @Inject
+    @InjectFromRole(value = FluxRole.ORCHESTRATION)
     public DbClearRule dbClearRule;
 
-    @Inject
+    @InjectFromRole(value = FluxRole.ORCHESTRATION)
     SimpleWorkflow simpleWorkflow;
 
-    @Inject
-    OrderedComponentBooter orderedComponentBooter;
+    @InjectFromRole(value = FluxRole.ORCHESTRATION)
+    OrchestrationOrderedComponentBooter orchestrationOrderedComponentBooter;
 
-    /** Needed to populate deployment units before beginning the test */
-    @Inject
+    /**
+     * Needed to populate deployment units before beginning the test
+     */
+    @InjectFromRole(value = FluxRole.EXECUTION)
     DeploymentUnitsManager deploymentUnitManager;
 
-    @Inject
-    @ManagedEnv
-    ExecutableRegistry registry;
+    @InjectFromRole(value = FluxRole.EXECUTION)
+    TaskExecutableRegistryImpl registry;
+
+    @InjectFromRole(value = FluxRole.EXECUTION)
+    ExecutionOrderedComponentBooter executionOrderedComponentBooter;
+
+    @Before
+    public void setUp() {
+        try {
```

Comment:
What is the purpose of this ?  Shouldn't these api call be covered as test in separate class.